R/calibrate_sim.R

```diff
@@ -0,0 +1,184 @@
+#'
+#'@description Starts a calibration run with multiple iterations using a GLM-AED2 setup. At the end a output csv.-file of all iterations and a 
+#'diagnostic plot are created.
+#'@param var Character vector of valid variable names (see \code{\link{sim_vars}})
+#'@param path String with the path to the GLM setup
+#'@param field.file CSV or TSV field data (see \link{resample_to_field}for format)
+#'@param nml.file String of the glm-namelist file, default is 'glm3.nml'
+#'@param calib_setup Data frame containing information regarding the calibration (see \link{get_calib_setup})
+#'@param glmcmd String containing the desired glm run command, default is GLM3r
+#'@param first.attempt Boolean, if TRUE a nml-template will be created, set it to FALSE after your first calibration run; default is TRUE
+#'@param period List that provides a start and a stop date for the simulation
+#'@param scaling Boolean, if TRUE variable values will be scaled on the space (0,10), recommended for CMA-ES, default is TRUE
+#'@param method String of the optimization method, default is 'CMA-ES' (Hansen 2009), alternatively you can also use 'Nelder-Mead'
+#'@param metric String of the calibration fit metric, default is RMSE
+#'@param target.fit Double of your preferred fit, calibration will stop after reaching that; default is 1.5
+#'@param target.iter Double of maximum amount of iterations, default is 150
+#'@param plotting Boolean, if TRUE plots all results as heat maps
+#'@param output Character array of the output folder path 
+#'@keywords methods
+#'@seealso \code{\link{get_calib_setup}}, \code{\link{get_calib_periods}}, \code{\link{get_calib_init_validation}}
+#'@author
+#'Robert Ladwig, Tadhg Moore
+#'@examples
+#'calib_setup <- get_calib_setup()
+#'print(calib_setup)
+#'
+#'#Example calibration
+#'#Copy files into temporary directory
+#'sim_folder <- tempdir() #simulation path
+#'glmtools_folder = system.file('extdata', package = 'glmtools')
+#'
+#'file.copy(list.files(glmtools_folder,full.names = TRUE), sim_folder, overwrite = TRUE)
+#'
+#'field.file <- file.path(sim_folder, 'LakeMendota_field_data_hours.csv')
+#'nml.file <- file.path(sim_folder, 'glm3.nml')
+#'driver_file <- file.path(sim_folder, 'LakeMendota_NLDAS.csv')
+#'period = get_calib_periods(nml = nml.file, ratio = 1)
+#'output = file.path(sim_folder, 'output/output.nc')
+#'
+#'var = 'temp' # variable to apply the calibration procedure
+#'calibrate_sim(var = var, path = sim_folder,
+#'              nml.file = nml.file, calib_setup = calib_setup,
+#'              glmcmd = NULL,
+#'              first.attempt = TRUE, period = period, method = 'CMA-ES',
+#'              scaling = TRUE, #scaling should be TRUE for CMA-ES
+#'              metric = 'RMSE',plotting = TRUE,
+#'              target.fit = 1.5,
+#'              target.iter = 50, output = output, field.file = field.file)
+#'@import adagio
+#'@import GLM3r 
+#'@import ggplot2
+#'@export
+calibrate_sim <- function(var = 'temp',
```

Comment:
Lots of cool stuff in here, but I wonder if there is too much? Perhaps there should be several functions, including one for plotting. Also, would it be better to use `verbose` as an argument that lets you turn on/off the print messages?

Reply:
Good idea on verbose. This has been added.